src/nix/flake.cc

```diff
@@ -1160,7 +1166,7 @@ struct CmdFlakeShow : FlakeCommand, MixJSON
 
     void run(nix::ref<nix::Store> store) override
     {
-        evalSettings.enableImportFromDerivation.setDefault(false);
+        evalSettings.enableImportFromDerivation.setDefault(allowImportFromDerivation);
```

Comment:
Afaik if this line was dropped, than allow-import-from-derivation would already default to true.